scip.proto

```diff
@@ -184,8 +184,59 @@ message SymbolInformation {
   repeated string documentation = 3;
   // (optional) Relationships to other symbols (e.g., implements, type definition).
   repeated Relationship relationships = 4;
+  // The kind of this symbol. Use this field instead of
+  // `SymbolDescriptor.Suffix` to determine whether something is, for example, a
+  // class or a method.
+  Kind kind = 5;
+  // Kind determines the type of a symbol using idiomatic names from the
+  // language's type system. For example, a Java method would have the kind
```

Comment:
This wording is a bit confusing since the "kind" is not the usual "kind" in type system jargon, "type" used as "category" again not in the sense of type systems, and yet there is a mention of a type system at the end of the sentence.

I would change it something like:

> // Kind represents the fine-grained category of a symbol, suitable for presenting
> // information about the symbol's meaning in the language.

(That's a bit too vague perhaps, but I would avoid using the words "type" and "type system" here.)

Maybe there should be an extra: "Since Kind is more fine-grained than Suffix, if two symbols have the same Kind, they should share the same Suffix. Conversely, if two symbols have different Suffixes, they should have different Kinds".

Reply:
> I would avoid using the words "type" and "type system" here

Agreed. The docstring here was written sloppily.

I like your proposed reformulation, and I think it's a good idea to mention the expected relationships between suffix and kinds.